### PR TITLE
BigQuery Labels

### PR DESCRIPTION
**Pull request by kconvey**
Implement BigQuery labels as a configurable option for the BQ adapter.

Also cleaned up Jinja whitespace a bit around 'create or replace' and 'OPTIONS', as there was a pretty huge block of empty lines in the query ultimately submitted to BQ.

#1942

### CODE REVIEW

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql [hunk 1]
```diff
@@ -27,41 +27,50 @@
 {%- endmacro -%}
 
 {% macro bigquery_table_options(persist_docs, temporary, kms_key_name) %}
-  {% set opts = {} %}
+  {% set opts = {} -%}
```
Comment (drewbanin) [inline plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: I think you're going to need to add `labels` to the macro signature here! Is `labels` on line 42 below currently defined?

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql [hunk 2]
```diff
@@ -27,41 +27,50 @@
...
-  {% if kms_key_name %}
-    {% do opts.update({'kms_key_name': "'" ~ kms_key_name ~ "'"}) %}
-  {% endif %}
+  {%- endif -%}
+  {%- if kms_key_name -%}
+    {%- do opts.update({'kms_key_name': "'" ~ kms_key_name ~ "'"}) -%}
+  {%- endif -%}
+  {%- if labels -%}
+    {%- set label_list = [] -%}
+    {%- for key in labels -%}
```
Comment (drewbanin) [inline plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: how about something like?
```suggestion
    {%- for label, value in labels.items() -%}
```

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql (hunk 1 above)
Comment (kconvey) [reply plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: Good catch!

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql (hunk 2 above)
Comment (kconvey) [reply plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: This is less ugly :)

**Comment by kconvey** [comment]
Added an integration test and incorporated your suggestions @drewbanin

**Comment by drewbanin** [comment]
thanks @kconvey! I kicked off the tests here. Will follow up when they finish running :)

**Comment by kconvey** [comment]
@drewbanin May need you to kick the integration tests off again. Made some tweaks to (hopefully) fix the integration test I added and add label support for views.

**Comment by drewbanin** [comment]
ok - tests should be running again now :)

Unsure if you saw [this comment](https://github.com/fishtown-analytics/dbt/issues/1942#issuecomment-560571013) but it looks like labels can also be supplied without values as "tags". Sounds like that pattern will work just fine with this implementation.

I need to do some smoke testing here, but largely, this is looking very good to me :D

**Comment by kconvey** [comment]
@drewbanin Did see that comment and I think the way that works in this implementation is that, since some labels have values, and yml dicts require that all keys have values, you have to set the empty string as the value for a tag when configuring in dbt.

ex. 
```
models:
  fishtown:
    materialized: table
    labels: {test_tag: '', test_label: 'test_value'}
```

This is consistent with the bq ddl syntax for tags:
```
ALTER TABLE mydataset.mytable
 SET OPTIONS (
   labels=[("tag1", ""), ("tag2", "")]
 )
```

(Taken from https://cloud.google.com/bigquery/docs/adding-labels#adding_a_tag)

**Comment by drewbanin** [comment]
merging this - thanks again :)

**Pull request merged**